### PR TITLE
Use cached features in /rows

### PR DESCRIPTION
**Pull request by lhoestq**
/rows needs the cached `features` since they're not always available in the parquet metadata.
This was causing some `Image` columns to be seen as a struct of binary data, which are not supported in the viewer (shown as "null").

Therefore I'm now passing the `features` from `config-parquet-and-info` to `config-parquet` and then to `config-parquet-metadata`. I kept it backward compatible in case a cached value doesn't have this field yet.

Therefore there's no need for a mongo migration. We can just re-run all the `config-parquet` and `config-parquet-metadata` jobs. I incremented their versions.

close https://github.com/huggingface/datasets-server/issues/1421

### CODE REVIEW

File: services/worker/src/worker/job_runners/config/parquet_metadata.py [hunk 1]
```diff
@@ -95,6 +95,11 @@ def compute_parquet_metadata_response(
         ]
         if not parquet_file_items:
             raise ParquetResponseEmptyError("No parquet files found.")
+        content = config_parquet_best_response.response["content"]
+        if "features" in content and isinstance(content["features"], dict):
+            features = content["features"]  # config-parquet version<6 didn't have features
+        else:
+            features = None
```
Comment (severo) [inline services/worker/src/worker/job_runners/config/parquet_metadata.py:102]: same comment:

maybe add a note to replace this part (features=None) later with an error. I understand you put it for backward-compatibility.

File: services/worker/tests/job_runners/config/test_parquet.py
Comment (severo) [inline services/worker/tests/job_runners/config/test_parquet.py:1]: maybe we could test the other branch of the if/else

File: services/rows/src/rows/routes/rows.py [hunk 2]
```diff
@@ -48,7 +48,7 @@
 ALL_COLUMNS_SUPPORTED_DATASETS_ALLOW_LIST: Union[Literal["all"], List[str]] = ["arabic_speech_corpus"]  # for testing
 
 # audio still has some errors when librosa is imported
-UNSUPPORTED_FEATURES_MAGIC_STRINGS = ["'binary'", "Audio("]
+UNSUPPORTED_FEATURES = [Value("binary"), Audio()]
```
Comment (severo) [inline services/rows/src/rows/routes/rows.py:51]: good! it's much more robust

File: libs/libcommon/src/libcommon/parquet_utils.py [hunk 3]
```diff
@@ -30,6 +31,18 @@ class FileSystemError(Exception):
     pass
 
 
+def _clean_mongo_objects(obj: Any) -> Any:
```
Comment (severo) [inline libs/libcommon/src/libcommon/parquet_utils.py:34]: maybe move this to the root function in simple_cache.py that gets the content out of the MongoDB?

File: libs/libcommon/src/libcommon/parquet_utils.py [hunk 4]
```diff
@@ -43,17 +56,28 @@ class ParquetFileMetadataItem(TypedDict):
...
+        def classify(feature: FeatureType) -> None:
+            nonlocal supported
+            for unsupported_feature in unsupported_features:
+                if type(unsupported_feature) == type(feature) == Value:
+                    if unsupported_feature.dtype == feature.dtype:
+                        supported = False
+                elif type(unsupported_feature) == type(feature):
+                    supported = False
+
+        _visit(feature, classify)
```
[the remark this answers is not in the document]
Comment (severo) [reply libs/libcommon/src/libcommon/parquet_utils.py:76]: excellent!

**Review by severo — APPROVED** [review]

**Comment by severo** [comment]
also: I think we have to update the openapi spec

File: services/worker/src/worker/job_runners/dataset/parquet.py [hunk 5]
```diff
@@ -83,7 +83,9 @@ def compute_parquet_response(dataset: str) -> Tuple[DatasetParquetResponse, floa
                 )
                 continue
             config_parquet_content = ConfigParquetResponse(
-                parquet_files=response["content"]["parquet_files"], partial=response["content"]["partial"]
+                parquet_files=response["content"]["parquet_files"],
+                partial=response["content"]["partial"],
+                features=None,
```
[the remark this answers is not in the document]
Comment (lhoestq) [reply services/worker/src/worker/job_runners/dataset/parquet.py:88]: `config_parquet_content` is instantiated for mypy typing I believe, but we don't return the features anyway in `DatasetParquetResponse` so no need to specify it.

I'll add a comment

**Pull request merged**